src/libtiled/hexagonalrenderer.cpp

```diff
@@ -67,23 +67,52 @@ QRect HexagonalRenderer::mapBoundingRect() const
 {
     const RenderParams p(map());
 
+    QRect mapBounds;
+
+    LayerIterator iterator(map());
+    while (Layer *layer = iterator.next()) {
+        if (TileLayer *tileLayer = dynamic_cast<TileLayer*>(layer))
+            mapBounds = mapBounds.united(tileLayer->bounds());
+    }
+
+    if (mapBounds.size() == QSize(0, 0))
+        mapBounds.setSize(QSize(1, 1));
+
     // The map size is the same regardless of which indexes are shifted.
     if (p.staggerX) {
-        QSize size(map()->width() * p.columnWidth + p.sideOffsetX,
-                   map()->height() * (p.tileHeight + p.sideLengthY));
+        if (!map()->infinite()) {
```

Comment:
Please consider that infinite and finite maps are not so different that you couldn't use the same code for both, just using the bounds:
```c++
QRect mapBounds;

if (map()->inifinite()) {
    // unite with the layer bounds
} else {
    mapBounds = QRect(0, 0, map()->width(), map()->height());
}

// Calculate bounding rect based on bounds (no need for checking infinite anymore)
```